### PR TITLE
Combine R-CMD-check and check-no-suggests.yaml

### PR DESCRIPTION
**Pull request by marcusmunch**
### Intent
The two workflow files do mainly the same thing, and to simplify things a little bit more, this PR combines the two.

Additionally, this PR restricts the jobs to only run if any files in `R/` or `tests/` are changed; it makes no sense to run `R CMD check` on change to, say, workflow definitions.

### Approach
* The job defined in `check-no-suggests.yaml` have in large part been moved to `R-CMD-check.yaml`
* jobs.R-CMD-check-hard.matrix only had one item defined, and job configuration has been changed to drop the matrix strategy

### Known issues
N/A
> If any known issues are introduced by this pull request, please let us know.
> Code is rarely perfect, and a pull request is still welcome if the fix is
> bigger than the new bug.

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -2,9 +2,15 @@
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
   push:
-    branches: [main, master]
+    branches: main
+    paths: |
+      - 'R/**'
+      - 'tests/**'
```
Comment (RasmusSkytte) [inline .github/workflows/R-CMD-check.yaml:8]: Consider adding a few more paths.
"man" -- sometimes we get stray .Rd files with commits that they are not connected to. 
If there are changes to .Rd files, we should still run the tests that check the .Rd files are formatted correctly (test-return.R, test-examples.R, and test-non_ascii.R)

"vignettes" -- updates to the vignettes should still be checked with R-CMD-Check.
```suggestion
      - 'tests/**'
      - 'man/**'
      - 'vignettes/**'
```

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
   pull_request:
-    branches: [main, master]
+    branches: main
+    paths: |
+      - 'R/**'
+      - 'tests/**'
```
Comment (RasmusSkytte) [inline .github/workflows/R-CMD-check.yaml:13]: Consider adding a few more paths.
"man" -- sometimes we get stray .Rd files with commits that they are not connected to. 
If there are changes to .Rd files, we should still run the tests that check the .Rd files are formatted correctly (test-return.R, test-examples.R, and test-non_ascii.R)

"vignettes" -- updates to the vignettes should still be checked with R-CMD-Check.
```suggestion
      - 'tests/**'
      - 'man/**'
      - 'vignettes/**'
```

**Review by RasmusSkytte — CHANGES_REQUESTED** [review]

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (marcusmunch) [reply .github/workflows/R-CMD-check.yaml:8]: Good point; I have committed the changes.

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (marcusmunch) [reply .github/workflows/R-CMD-check.yaml:13]: Good point; I have committed the changes.

**Comment by marcusmunch** [comment]
The deploy job failing is fixed as of a60ebb67628598827fa5774d43f58d983fd81674, but GitHub does not (at least easily) allow for re-running workflows with updated definition files, and this check may therefore be disregarded.

**Review by RasmusSkytte — APPROVED** [review]

**Pull request merged**